frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java

```diff
@@ -301,34 +301,45 @@ public void run() {
+                        if (parsedMetric == null) {
+                            logger.error("Failed to parse metrics line: \"{}\".", result);
+                            continue;
+                        }
+
+                        try {
                             if (this.metricCache.getMetricBackend(parsedMetric.getMetricName())
-                                    != null) {
-                                try {
-                                    List<String> dimensionValues = new ArrayList<String>();
-                                    for (Dimension dimension : parsedMetric.getDimensions()) {
-                                        dimensionValues.add(dimension.getValue());
-                                    }
-                                    // Hostname is added as a dimension by default to backend
-                                    // metrics
-                                    dimensionValues.add(parsedMetric.getHostName());
-                                    this.metricCache
-                                            .getMetricBackend(parsedMetric.getMetricName())
-                                            .addOrUpdate(
-                                                    dimensionValues,
-                                                    parsedMetric.getRequestId(),
-                                                    Double.parseDouble(parsedMetric.getValue()));
-                                } catch (Exception e) {
-                                    logger.error(
-                                            "Failed to update backend metric ",
-                                            parsedMetric.getMetricName(),
-                                            ": ",
-                                            e);
+                                    == null) {
+                                if (!lifeCycle.configManager.isModelMetricsAutoDetectEnabled()) {
+                                    continue;
                                 }
+
+                                logger.info(
+                                        "Registering auto detected backend metric: {}",
+                                        parsedMetric);
+                                this.metricCache.addAutoDetectMetricBackend(parsedMetric);
                             }
-                        } else {
-                            logger.error("Failed to parse metrics line: \"{}\".", result);
+
+                            List<String> dimensionValues = new ArrayList<String>();
+                            for (Dimension dimension : parsedMetric.getDimensions()) {
+                                dimensionValues.add(dimension.getValue());
+                            }
+                            // Hostname is added as a dimension by default to backend metrics
+                            dimensionValues.add(parsedMetric.getHostName());
+
+                            this.metricCache
```

Comment:
ditto

Reply:
Added details in this comment: https://github.com/pytorch/serve/pull/2769/files#r1399626654